src/UXClient/Components/LineChart/LineChart.ts

```diff
@@ -217,6 +233,118 @@ class LineChart extends ChartComponent {
+    }
+
+    private setScooterLabels (scooter, includeTransition = false) {
+        var millis = this.scooterGUIDMap[scooter.datum()];
+        var values = this.chartComponentData.timeMap[millis] != undefined ? this.chartComponentData.timeMap[millis] : [];
+        var self = this;
+
+        var valueLabels = scooter.selectAll(".tsi-scooterValue").data(values, (d) => {
+            return d.aggregateKey + "_" + d.splitBy;
+        });
+
+        valueLabels.enter()
+            .append("div")
+            .classed("tsi-scooterValue", true)
+            .merge(valueLabels)
+            .each(function (d: any) {
+                var valueLabel = d3.select(this).selectAll(".tsi-scooterValueLabel").data([d]);
+                valueLabel.enter()
+                    .append("div")
+                    .classed("tsi-scooterValueLabel", true)
+                    .merge(valueLabel)
+                    .html(() => Utils.formatYAxisNumber(self.getValueOfVisible(d)))
+                    .style("border-color", () => self.colorMap[d.aggregateKey + "_" + d.splitBy])
+                valueLabel.exit().remove();
+
+                var valueCaret = d3.select(this).selectAll(".tsi-scooterValueCaret").data([d])
+                valueCaret.enter()
+                    .append("div")
+                    .classed("tsi-scooterValueCaret", true)
+                    .merge(valueCaret)
+                    .style("border-right-color", () => self.colorMap[d.aggregateKey + "_" + d.splitBy]);
+                valueCaret.exit().remove();
+            })
+            .transition()
+            .duration((self.chartOptions.noAnimate || !includeTransition) ? 0 : self.TRANSDURATION)
+            .style("top", (d) => this.calcTopOfScooterValueLabel(d));
+
+        valueLabels.exit().remove();
+    }
+
+    private getValueOfVisible (d) {
+        if (d.measures) {
+            var visibleMeasure = this.chartComponentData.getVisibleMeasure( d.aggregateKey, d.splitBy);
+            if (d.measures[visibleMeasure])
+                return d.measures[visibleMeasure];
+        } 
+        return null;
+    }
+
+    // updates the display of scooters but not their underlying data. Updates all if no scooter is passed in
+    private updateScooterPresentation (scooter = null) {
+        var scooterSelection;
+        if (scooter != null) {
+            scooterSelection = scooter;
+        } else {
+            scooterSelection = d3.select(this.renderTarget).selectAll(".tsi-scooterContainer");
+        }
+        var self = this;
+        scooterSelection.each(function () {
+            var currScooter = d3.select(this);
+            self.setScooterLabels(currScooter, true);
+            currScooter.transition()
+                .duration(self.chartOptions.noAnimate ? 0 : self.TRANSDURATION)
+                .style("left", (d) => (Math.round(self.x(d) + self.getScooterMarginLeft()) + "px"));
+        });
+    } 
+
```

Comment:
generally a little concerned about the volume of the additions to line chart vs creating a scooter component, but i will defer to you on this decision